File: algorithms/implementation/happy-ladybugs/happy-ladybugs-java/happy-ladybugs-java/src/TesteHappyLadybugs.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class TesteHappyLadybugs {

    @Test
    public void testeCaso1() {

        String b = "RBY_YBR";

        String esperado = "YES";

        String resultado = Resultado.happyLadybugs(b);

        assertEquals(resultado, esperado);
    }

    @Test
    public void testeCaso2() {

        String b = "X_Y__X";

        String esperado = "NO";

        String resultado = Resultado.happyLadybugs(b);

        assertEquals(resultado, esperado);
    }
    
}

File: algorithms/implementation/happy-ladybugs/happy-ladybugs-javascript/js/testeHappyLadybugs.js
import {happyLadybugs} from '../js/happyLadybugs.js';
import assert from 'assert';

describe('happyLadybugs', function () {
   
    it('Deve retornar YES para uma string feliz', function () {
        

        let b = 'RBY_YBR';

        let esperado = 'YES';

        let resultado = happyLadybugs(b)

        assert.strictEqual(resultado, esperado)
    });
});


describe('happyLadybugs', function () {
   
    it('Deve retornar NO para uma string feliz', function () {
        

        let b = 'X_Y__X';

        let esperado = 'NO';

        let resultado = happyLadybugs(b)

        assert.strictEqual(resultado, esperado)
    });
});

describe('happyLadybugs', function () {
   
    it('Deve retornar YES para uma string feliz', function () {
        

        let b = '__';

        let esperado = 'YES';

        let resultado = happyLadybugs(b)

        assert.strictEqual(resultado, esperado)
    });
});